### PR TITLE
client stability fixes

### PR DESCRIPTION
**Pull request by willscott**
* Hopefully address #303 with some timeouts on startup
* warnings print more cleanly
* slack added to watch routine to make sure randomness is ready when polled
* client binary switched over to functional options

### CODE REVIEW

File: client/util.go [hunk 1]
```diff
@@ -9,6 +9,10 @@ import (
 	"github.com/drand/drand/log"
 )
 
+// How long to wait after the period begins to ask for the randomness from that period.
+// TODO: ongoing re-calibration of this value.
+const slack = time.Second
```
Comment (nikkolasg) [inline client/util.go:14]: Can you recall again why we aren't using the stream function instead of manually giving some slack ?

Comment (willscott) [reply client/util.go:14]: because you can't push over http.

File: client/util.go [hunk 2]
```diff
@@ -30,7 +34,7 @@ func pollingWatcher(ctx context.Context, client Client, group *key.Group, log lo
 		select {
 		case <-ctx.Done():
 			return
-		case <-time.After(time.Duration(nextTime-time.Now().Unix()) * time.Second):
+		case <-time.After(time.Duration(nextTime-time.Now().Unix())*time.Second + slack):
```
Comment (alanshaw) [inline client/util.go:37]:
```suggestion
		case <-time.After(time.Duration(nextTime-time.Now().Unix())*time.Second + math.Min(slack, group.Period)):
```

File: http/server_test.go [hunk 3]
```diff
@@ -27,6 +28,7 @@ func TestHTTPRelay(t *testing.T) {
 	defer cancel()
 
 	client := drand.NewPublicClient(conn)
+	time.Sleep(100 * time.Millisecond)
```
Comment (alanshaw) [inline http/server_test.go:31]: Urgh, we need to get rid of these. There should be a mechanism for us to know when the client is ready...

File: http/server_test.go [hunk 4]
```diff
@@ -40,6 +42,7 @@ func TestHTTPRelay(t *testing.T) {
 	server := http.Server{Handler: handler}
 	go server.Serve(listener)
 	defer server.Shutdown(ctx)
+	time.Sleep(100 * time.Millisecond)
```
Comment (alanshaw) [inline http/server_test.go:45]: Likewise, we need a way to know when the HTTP server has started.

File: http/server_test.go (hunk 3 above)
Comment (willscott) [reply http/server_test.go:31]: yep. agree :-| I'm somewhat unconvinced this fully solves the problem, but i want to see if this stops the occasional 'first group request seen on new server doesn't yet have a group initialized' heisenbug that's popped up on CI a few times.

**Comment by alanshaw** [comment]
ahh, you might need to import `math` and do some type converting...sorry!

**Pull request merged**